docs/CH-Edgex-taf.rst

```diff
@@ -0,0 +1,152 @@
+
+       cd edgex-taf
+       git submodule init
+       git submodule update --remote
+
+4. Edit your config and utils under TAF directory:
+
+   .. image:: images/Edgex-taf-TAF.png
+       :scale: 50%
+       :alt: Edgex-taf TAF directory
+
+5. Put the use cases or test cases in the TAF/testScenarios folder
+
+   Create a robot test case must contain the Settings, Variables and Keywords shown below
+
+   .. image:: images/Edgex-taf-Robot.png
+       :scale: 50%
+       :alt: Edgex-taf Robot Test Case
+
+6. Put the test scripts in the TAF/testCaseApps folder
+
+7. Run the tests using the following commands::
+
+       python3 edgex-taf-common/TAF-Manager/trigger/run.py -u '*'
+
+Configuration Profiles
+
+Edgex-taf defines the configuration folder separated by different profiles; the user can trigger the testing using the profile name. 
+
+For example, to run the certification testing for **device-random**::
+
+    python3 edgex-taf-common/TAF-Manager/trigger/run.py -p device-random -u '*'
+
+Or to run the certification testing for **device-modbus**::
+
+    python3 edgex-taf-common/TAF-Manager/trigger/run.py -p device-modbus -u '*'
+
+To run the default certification testing::
+
+    python3 edgex-taf-common/TAF-Manager/trigger/run.py -u '*'
+
+.. image:: images/Edgex-taf-Workspace.png
+    :scale: 50%
+    :alt: Edgex-taf Workspace
+
+Python System Path Setup
+========================
+
+We use the project root path and edgex-taf-common as the system path. The python module names are **TAF** and **TUC**. This is illustrated below:
+
+.. image:: images/Edgex-taf-working-dir.png
+    :scale: 50%
+    :alt: Edgex-taf Working Directory
+
+|
+
+.. image:: images/Edgex-taf-TAF-settings.png
+    :scale: 50%
+    :alt: Edgex-taf TAF Settings
+
+|
+
+.. image:: images/Edgex-taf-Import.png
+    :scale: 50%
+    :alt: Edgex-taf TUC Import
+
+Using the EdgeX-taf-common docker Container
+===========================================
+
+Trigger the testing using edgex-taf-common/TAF-Manager/trigger/run.py. It must install the robotframework and required tools; we built a docker image to wrap all required tools, so that the user could trigger testing without installing any dependencies. Users only need to mount the testing scripts and trigger the testing.
+
+.. image:: images/Edgex-taf-common.png
+    :scale: 50%
+    :alt: Edgex-taf Container
+
+Build the edgex-taf-common docker Image
+---------------------------------------
+
+1. Use git to clone edgex-taf-common::
+
+       git clone https://github.com/edgexfoundry-holding/edgex-taf-common.git
+
+2. Change the directory to edgex-taf-common and execute the command::
+
+       cd edgex-taf-common
+       docker build -t docker-edgex-taf-common:1.1.1 .
+
+3. Push to the docker registry::
+
+       docker push docker-edgex-taf-common:1.1.1
+
+Run the Tests
+=============
+
+Run the complete test suite as follows::
```

Comment:
Where is the entry point to the TAF ROBOT code when docker run command below is issued.

Reply:
We create the docker image via edgex-taf-common https://github.com/edgexfoundry/edgex-taf-common and you can see the entrypoint file also there.